### PR TITLE
feat(metrics): add Amplitude events for subscription upgrades

### LINKED ISSUE #3274: Track amplitude metrics for subscription upgrades
We will likely need to include some tracking for amplitude of product upgrades.

@irrationalagent Any input on what data we might want to include or how to represent a product/plan upgrade?

### PR DESCRIPTION
**Pull request by chenba**
Fixes #3274 

@mozilla/fxa-devs r?

### CODE REVIEW

File: packages/fxa-payments-server/src/components/PaymentForm/index.tsx [hunk 1]
```diff
@@ -283,7 +283,7 @@ const validateZip: OnValidateFunction = (value, focused) => {
   } else if (value.length !== 5 && !focused) {
     valid = false;
     error = 'Zip code is too short';
-  } else if (value.length == 5) {
+  } else if (value.length === 5) {
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/components/PaymentForm/index.tsx]: Huh, I thought we'd fixed that in another PR already. That lint warning's been around too long.  😞

**Comment by bbangert** [comment]
The test error is a node 10 vs 12 issue. It looks like the script wants Node 10. I thought this was fixed though, maybe rebase against latest master?

**Comment by lmorchard** [comment]
Oof, yeah, this is a different failure [than last time](https://circleci.com/gh/mozilla/fxa/128914). Fun!

**Comment by chenba** [comment]
> maybe rebase against latest master?

Yeah, that should fix it.

**Pull request merged**